src/go/rpk/pkg/cli/cmd/cluster/config/export/export.go

```diff
@@ -0,0 +1,205 @@
+				Kind:  yaml.ScalarNode,
+				Value: x,
+			}
+		case bool:
+			var x_str string
+			if x {
+				x_str = "true"
+			} else {
+				x_str = "false"
+			}
+			valueNode = yaml.Node{
+				Kind:  yaml.ScalarNode,
+				Value: x_str,
+			}
+		case []interface{}:
+			valueNode = yaml.Node{
+				Kind:  yaml.SequenceNode,
+				Value: "",
+			}
+			// TODO generalize for other types of array
+			for _, v := range x {
+				seqNode := yaml.Node{
+					Kind:  yaml.ScalarNode,
+					Value: v.(string),
+				}
+				valueNode.Content = append(valueNode.Content, &seqNode)
+			}
+		case nil:
+			valueNode = yaml.Node{
+				Kind:  yaml.ScalarNode,
+				Value: "",
+			}
+		default:
+			log.Warningf("Unexpected property value type: %s: %T", name, curValue)
+			continue
+		}
+
+		// TODO: nullability in description
+		// TODO: needs-restart in description
+
+		// Preface each property with a descriptive comment
+		if meta.Example != "" {
+			property_node.HeadComment = fmt.Sprintf("%s (e.g. '%s')", meta.Description, meta.Example)
+		} else {
+			property_node.HeadComment = meta.Description
+		}
+
+		// Blank line between each property for readability
+		property_node.FootComment = "\n"
+
+		yamlNode.Content = append(yamlNode.Content, &property_node)
+		yamlNode.Content = append(yamlNode.Content, &valueNode)
+	}
+
+	serialized, err := yaml.Marshal(&yamlNode)
+	if err != nil {
+		return err
+	}
+
+	_, err = file.Write(serialized)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func NewCommand(fs afero.Fs, all *bool) *cobra.Command {
+	var filename string
+
+	cmd := &cobra.Command{
+		Use:   "export",
+		Short: "Export cluster configuration.",
+		Long: `Export cluster-wide configuration properties
+
+Cluster-wide properties apply to all nodes in a redpanda cluster.  For
+node properties, use the 'rpk config' command.
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			client, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			// GET the schema
+			schema, err := client.ClusterConfigSchema()
+			out.MaybeDie(err, "unable to query config schema: %v", err)
+
+			// GET current config
+			var currentConfig admin.Config
+			currentConfig, err = client.Config()
+			out.MaybeDie(err, "unable to query current config: %v", err)
+
+			// Generate a yaml template for editing
+			var file *os.File
+			if filename == "" {
+				file, err = ioutil.TempFile("/tmp", "config_*.yaml")
+			} else {
+				file, err = os.Create(filename)
+			}
+			out.MaybeDie(err, "unable to create temporary file: %v", err)
+			err = ExportConfig(file, schema, currentConfig, *all)
+			out.MaybeDie(err, "Failed to write out config: %v", err)
+			err = file.Close()
+			log.Infof("Wrote configuration to file %s", file.Name())
+			file = nil
+			out.MaybeDie(err, "error closing temporary file: %v", err)
```

Comment:
This is only a temporary file is the user did not specify a file. While in a strict sense it's good practice to check file, it really shouldn't error unless there's filesystem corruption, and often you don't see the file close error checked / logged. If we keep this check&log, I think the log should be `error closing file %q: %v", file.Name(), err)` but I'd remove the check altogether personally.

Reply:
[grizzled filesystem guy voice...] close() can return an error in non-corruption cases on network filesystems (e.g. many home directories where someone might export a config).  In some cases it can also error out on an enospc, if a buffered write was done, then couldn't be synced, and the only place to bubble that up to the user is on close.

But the message was indeed stale here (it originally was written in the `edit` path), fixing that.